References
----------

.. [Amir13] Amir *et al.* (2013),
   *viSNE enables visualization of high dimensional single-cell data and reveals phenotypic heterogeneity of leukemia*,
   `Nature Biotechnology <https://doi.org/10.1038/nbt.2594>`_.

.. [Angerer16] Angerer *et al.* (2016),
   *destiny – diffusion maps for large-scale single-cell data in R*,
   `Bioinformatics <https://doi.org/10.1093/bioinformatics/btv715>`_.

.. [Blondel08] Blondel *et al.* (2008),
   *Fast unfolding of communities in large networks*,
   `J. Stat. Mech. <https://doi.org/10.1088/1742-5468/2008/10/P10008>`_.

.. [Chippada18]
   *ForceAtlas2 for Python and NetworkX*,
   `GitHub <https://github.com/bhargavchippada/forceatlas2>`_.

.. [Coifman05] Coifman *et al.* (2005),
   *Geometric diffusions as a tool for harmonic analysis and structure definition of data: Diffusion maps*,
   `PNAS <https://doi.org/10.1038/nmeth.3971>`_.

.. [Csardi06] Csardi *et al.* (2006),
   *The igraph software package for complex network researc*,
   `InterJournal Complex Systems <http://igraph.org>`_.

.. [Ester96] Ester *et al.* (1996),
   *A Density-Based Algorithm for Discovering Clusters in Large Spatial Databases with Noise*,
   `Proceedings of the 2nd International Conference on Knowledge Discovery and Data Mining,
   Portland, OR <http://citeseerx.ist.psu.edu/viewdoc/summary?doi=10.1.1.121.9220>`_.

.. [Eulenberg17] Eulenberg *et al.* (2017),
   *Reconstructing cell cycle and disease progression using deep learning*
   `Nature Communications <https://doi.org/10.1101/081364>`_.

.. [Fechtner18]
   *PyPairs*,
   `GitHub <https://github.com/rfechtner/pypairs>`_.

.. [Fruchterman91] Fruchterman & Reingold (1991),
   *Graph drawing by force-directed placement*,
   `Software: Practice & Experience <http://doi.org:10.1002/spe.4380211102>`_.

.. [Hagberg08] Hagberg *et al.* (2008),
   *Exploring Network Structure, Dynamics, and Function using NetworkX*,
   `Scipy Conference <http://conference.scipy.org/proceedings/SciPy2008/paper_2/>`_.

.. [Hastie09]
   Hastie *et al.* (2009),
   *The Elements of Statistical Learning*,
   `Springer <https://web.stanford.edu/~hastie/ElemStatLearn/>`_.

.. [Haghverdi15] Haghverdi *et al.* (2015),
   *Diffusion maps for high-dimensional single-cell analysis of differentiation data*,
   `Bioinformatics <https://doi.org/10.1093/bioinformatics/btv325>`_.

.. [Haghverdi16] Haghverdi *et al.* (2016),
   *Diffusion pseudotime robustly reconstructs branching cellular lineages*,
   `Nature Methods <https://doi.org/10.1038/nmeth.3971>`_.

.. [Haghverdi18] Haghverdi *et al.* (2018),
   *Batch effects in single-cell RNA-sequencing data are corrected by matching mutual nearest neighbors*,
   `Nature Biotechnology <10.1038/nbt.4091>`_.

.. [Huber15] Huber *et al.* (2015),
   *Orchestrating high-throughput genomic analysis with Bioconductor*,
   `Nature Methods <https://doi.org/10.1038/nmeth.3252>`_.

.. [Islam11] Islam *et al.* (2011),
   *Characterization of the single-cell transcriptional landscape by highly multiplex RNA-seq*,
   `Genome Research <https://doi.org/10.1101/gr.110882.110>`_.

.. [Jacomy14] Jacomy *et al.* (2014),
   *ForceAtlas2, a Continuous Graph Layout Algorithm for Handy Network Visualization Designed for the Gephi Software*
   `PLOS One <https://doi.org/10.1371/journal.pone.0098679>`_.

.. [Kang18] Kang *et al.* (2018),
   *Python Implementation of MNN correct*,
   `GitHub <https://github.com/chriscainx/mnnpy>`_.

.. [Krumsiek10] Krumsiek *et al.* (2010),
   *Odefy – From discrete to continuous models*,
   `BMC Bioinformatics <https://doi.org/10.1186/1471-2105-11-233>`_.

.. [Krumsiek11] Krumsiek *et al.* (2011),
   *Hierarchical Differentiation of Myeloid Progenitors Is Encoded in the Transcription Factor Network*,
   `PLoS ONE <https://doi.org/10.1371/journal.pone.0022649>`_.

.. [Levine15] Levine *et al.* (2015),
   *Data-Driven Phenotypic Dissection of AML Reveals Progenitor--like Cells that Correlate with Prognosis*,
   `Cell <https://doi.org/10.1016/j.cell.2015.05.047>`_.

.. [Maaten08] Maaten & Hinton (2008),
   *Visualizing data using t-SNE*,
   `JMLR <http://www.jmlr.org/papers/v9/vandermaaten08a.html>`_.

.. [McCarthy17] McCarthy *et al.* (2017),
   *scater: Single-cell analysis toolkit for gene expression data in R*,
   `Bioinformatics <https://doi.org/10.1093/bioinformatics/btw777>`_.

.. [Moon17] Moon *et al.* (2017),
   *PHATE: A Dimensionality Reduction Method for Visualizing Trajectory Structures in High-Dimensional Biological Data*,
   `BioRxiv <http://biorxiv.org/content/early/2017/03/24/120378>`_.

.. [Satija15] Satija *et al.* (2015),
   *Spatial reconstruction of single-cell gene expression data*,
   `Nature Biotechnology <https://doi.org/10.1038/nbt.3192>`_.

.. [McInnes18] McInnes & Healy (2018),
   *UMAP: Uniform Manifold Approximation and Projection for Dimension Reduction*,
   `arXiv <https://arxiv.org/abs/1802.03426>`_.

.. [Moignard15] Moignard *et al.* (2015),
   *Decoding the regulatory network of early blood development from single-cell gene expression measurements*,
   `Nature Biotechnology <https://doi.org/10.1038/nbt.3154>`_.

.. [Murphy12]
   Murphy (2012,
   *Machine Learning: A Probabilisitc Perspective*,
   `MIT Press <https://mitpress.mit.edu/books/machine-learning-0>`_.

.. [Ntranos18] Ntranos *et al.* (2018),
   *Identification of transcriptional signatures for cell types from single-cell RNA-Seq*,
   `bioRxiv <https://doi.org/10.1101/258566>`_.

.. [Pedregosa11] Pedregosa *et al.* (2011),
   *Scikit-learn: Machine Learning in Python*,
   `JMLR <http://www.jmlr.org/papers/v12/pedregosa11a.html>`_.

.. [Paul15] Paul *et al.* (2015),
   *Transcriptional Heterogeneity and Lineage Commitment in Myeloid Progenitors*,
   `Cell <https://doi.org/10.1016/j.cell.2015.11.013>`_.

.. [Scialdone15] Scialdone *et al.* (2015),
   *Computational assignment of cell-cycle stage from single-cell transcriptome data*
   `Methods <https://doi.org/10.1016/j.ymeth.2015.06.021>`_.

.. [Traag17] Traag (2017),
   *Louvain*,
   `GitHub <https://doi.org/10.5281/zenodo.35117>`_.

.. [Ulyanov16] Ulyanov (2016),
   *Multicore t-SNE*,
   `GitHub <https://github.com/DmitryUlyanov/Multicore-TSNE>`_.

.. [Weinreb17] Weinreb *et al.* (2016),
   *SPRING: a kinetic interface for visualizing high dimensional single-cell expression data*,
   `bioRxiv <https://doi.org/10.1101/090332>`_.

.. [Wittmann09] Wittmann *et al.* (2009),
   *Transforming Boolean models to continuous models: methodology and application to T-cell receptor signaling*,
   `BMC Systems Biology <https://doi.org/10.1186/1752-0509-3-98>`_.

.. [Waskom16] Waskom *et al.* (2017),
   *Seaborn*,
   `Zenodo <https://doi.org/10.5281/zenodo.54844>`_.

.. [Wolf17] Wolf *et al.* (2018),
   *Scanpy: large-scale single-cell gene expression data analysis*,
   `Genome Biology <https://doi.org/10.1186/s13059-017-1382-0>`_.

.. [Wolf17i] Wolf *et al.* (2017),
   *Graph abstraction reconciles clustering with trajectory inference through a topology preserving map of single cells.*
   `bioRxiv <https://doi.org/10.1101/208819>`_.

.. [Zheng17] Zheng *et al.* (2017),
   *Massively parallel digital transcriptional profiling of single cells*,
   `Nature Communications <https://doi.org/10.1038/ncomms14049>`_.

.. [Zunder15] Zunder *et al.* (2015),
   *A continuous molecular roadmap to iPSC reprogramming through progression analysis of single-cell mass cytometry*,
   `Cell Stem Cell <https://doi.org/10.1016/j.stem.2015.01.015>`_.
